game/scripts/Player.lua

```diff
@@ -18,19 +18,68 @@ function script:init()
 	self.turnSpeed = 200
 	self.timer = 0
     self.slowDown = 80
-    self.transform.position = vector(0, 2, 0)
-    self.transform.rotation = vector(-90, 0, 0)
+    self.transform.position = vector(0, 10, 0)
+    self.transform.rotation = vector(0, 0, 0)
 
     self.maxHP = 100.0
     self.currentHP = self.maxHP
+    self.onGround = false
+    self.active = true
 end
 
 function script:update(dt)
-    self:move2(dt)
-    self:rotate2(dt)
+    -- Set and use active
+    if (input.isKeyPressed(Keys.ESCAPE)) then
+        self.active = not self.active
+    end
+
+    if (not self.active) then
+        return
+    end
+
+    -- Check if grounded
+    local payload = physics.raycast(self.transform.position, vector(0, -1, 0))
+    if (payload) then
+        self.onGround = (self.transform.position - payload.hitPoint):length() < 5
+    end
+
+    -- New movement using rigidbody
+    local camTransform = scene.getComponent(self.camID, CompType.Transform)
+    local forward = camTransform:forward()
+    forward.y = 0
+    local right = camTransform:right()
+    right.y = 0
+
+    self.moveDir = vector(core.btoi(input.isKeyDown(Keys.A)) - core.btoi(input.isKeyDown(Keys.D)), core.btoi(input.isKeyDown(Keys.W)) - core.btoi(input.isKeyDown(Keys.S)), 0)
+    self.currentSpeed = self.moveDir:normalize() * self.maxSpeed * math.max(1.0, self.perkProperties.speedPerkActive * self.perkProperties.speedPerkValue)
+    self.currentSpeed = forward:normalize() * self.currentSpeed.y + right:normalize() * self.currentSpeed.x
```

Comment:
Would have been nice to have a comment explaining line 54 is the 2D speed/direction in local space, while line 55 is transforming that vector to 3D world space.